rclcpp/src/rclcpp/parameter_value.cpp

```diff
@@ -189,12 +193,16 @@ ParameterValue::ParameterValue(const std::vector<bool> & bool_array_value)
 ParameterValue::ParameterValue(const std::vector<int> & int_array_value)
 {
   value_.integer_array_value.assign(int_array_value.cbegin(), int_array_value.cend());
+  // Allow integer arrays to be cast to double arrays
+  value_.double_array_value.assign(int_array_value.cbegin(), int_array_value.cend());
```

Comment:
I'm not a fan of this. For primitive types it is not as bad (though I still don't like the idea that the value has two values stored), but for the array I think there's a chance that could be lots of overhead.